test/prometheus_diameter_collector_SUITE.erl

```diff
@@ -0,0 +1,135 @@
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(prometheus_diameter_collector_SUITE).
+
+%%% ==================================================================
+%%% Common Tests Callbacks Exports
+%%% ==================================================================
+
+-export([
+    all/0,
+    groups/0,
+    init_per_suite/1,
+    end_per_suite/1
+]).
+
+%%% ==================================================================
+%%% Common Tests Exports
+%%% ==================================================================
+
+-export([
+    diameter_applications/1,
+    diameter_connections/1,
+    diameter_messages/1,
+    gather_statistics/1
+]).
+
+%%% ==================================================================
+%%% Includes
+%%% ==================================================================
+
+-include_lib("common_test/include/ct.hrl").
+
+%%% ==================================================================
+%%% Macros
+%%% ==================================================================
+
+-define(match(Guard, Expr),
+    ((fun () ->
+        case (Expr) of
+            Guard ->
+                ok;
+            V ->
+                ct:pal("MISMATCH(~s:~b, ~s)~nExpected: ~p~nActual:   ~p~n",
+                [?FILE, ?LINE, ??Expr, ??Guard, V]),
+                error(badmatch)
+        end
+    end)())).
+
+%%% ==================================================================
+%%% Common Tests Callbacks
+%%% ==================================================================
+
+all() ->
+  [
+    {group, collector}
+  ].
+
+groups() ->
+    [
+        {collector, [sequence], [
+            diameter_applications,
+            diameter_connections,
+            diameter_messages,
+            gather_statistics
+        ]}
+    ].
+
+init_per_suite(Config) ->
+    ok = diameter:start(),
+    ok = prometheus:start(),
+    ok = prometheus_registry:register_collector(pdc, prometheus_diameter_collector),
+    Config.
+
+end_per_suite(Config) ->
+    ok = diameter:stop(),
+    ok = prometheus:stop(),
+    Config.
+
+%%% ==================================================================
+%%% Common Tests
+%%% ==================================================================
+
+%% -------------------------------------------------------------------
+%% @doc
+%% Check diameter applications
+%% @end
+%% -------------------------------------------------------------------
```

Comment:
CT suite usually use a info callback to a test to supply documentation.

```suggestion
diameter_applications() ->
    [{doc, "Check diameter applications"}].
```

Reply:
Done, thanks.